### PR TITLE
fix: add pair token price handler

### PR DESCRIPTION
**Pull request by berteotti**

### CODE REVIEW

File: src/commons/addresses.ts [hunk 1]
```diff
@@ -105,3 +105,79 @@ export function getLiquidityTrackingTokenAddresses(): string[] {
...
+ * Swapr token address, supported network: Mainnet, Arbitrum One, xDAI, Rinkeby, and Arbitrum Rinkeby
+ */
+export function getSwaprTokenAddress(): string {
+  let network = dataSource.network() as string
+  // Production
+  if (network == 'mainnet') return '0x6cacdb97e3fc8136805a9e7c342d866ab77d0957'
+  if (network == 'arbitrum-one') return '0xde903e2712288a1da82942dddf2c20529565ac30'
+  if (network == 'xdai') return '0x532801ED6f82FFfD2DAB70A19fC2d7B2772C4f4b'
+  /**
+   * @todo get the correct addreses
```
Comment (guerrap) [inline src/commons/addresses.ts]: Why this todo?

File: src/commons/addresses.ts [hunk 2]
```diff
@@ -105,3 +105,79 @@ export function getLiquidityTrackingTokenAddresses(): string[] {
...
+  log.warning('no Swapr address for unsupported network {}', [network])
+  return ADDRESS_ZERO
+}
+
+/**
+ * Checks if the token address is a Swapr token
+ * @param address
+ */
+export function isSwaprToken(address: Address): boolean {
+  // let network = dataSource.network() as string;
```
Comment (guerrap) [inline src/commons/addresses.ts]: This can be removed right?

File: src/commons/addresses.ts [hunk 3]
```diff
@@ -105,3 +105,79 @@ export function getLiquidityTrackingTokenAddresses(): string[] {
...
+  // for now, treat everything as true value
+  return address.toHexString() == getSwaprTokenAddress()
+}
+
+/**
+ * Checks if the token address is a Swapr LP token
+ * @param address
+ */
+export function isSwaprLPToken(address: Address): boolean {
+  // let network = dataSource.network() as string;
```
Comment (guerrap) [inline src/commons/addresses.ts]: This too.

File: src/mappings/core.ts [hunk 4]
```diff
@@ -43,3 +47,73 @@ export function handlePairTokenPrice(block: ethereum.Block): void {
...
   }
 }
+
+export function handleSync(event: Sync): void {
+  let pair = Pair.load(event.address.toHex()) as Pair
+  let token0 = Token.load(pair.token0) as Token
+  let token1 = Token.load(pair.token1) as Token
+  let swapr = getSwaprFactory()
+
+  // reset factory liquidity by subtracting onluy tarcked liquidity
```
Comment (guerrap) [inline src/mappings/core.ts]: Small typos.

File: subgraph.rinkeby.yaml [hunk 5]
```diff
@@ -36,7 +36,7 @@ dataSources:
 templates:
   - kind: ethereum/contract
     name: Pair
-    network: rinkeby
+    network: mainnet
```
Comment (guerrap) [inline subgraph.rinkeby.yaml]: Why `mainnet`?

**Review by guerrap — CHANGES_REQUESTED** [review]

File: src/commons/addresses.ts (hunk 1 above)
Comment (berteotti) [reply src/commons/addresses.ts]: It was copied from the original subgraph repo. It's fixed now :)

File: subgraph.rinkeby.yaml (hunk 5 above)
Comment (berteotti) [reply subgraph.rinkeby.yaml]: Typo, just fixed it!

**Review by guerrap — APPROVED** [review]

**Pull request merged**